activation/activation_test.go

```diff
@@ -1113,3 +1078,51 @@ func TestBuilder_UpdatePoetsUnstable(t *testing.T) {
 	r.ErrorIs(err, ErrPoetServiceUnstable)
 	r.Nil(tab.receivePendingPoetClients())
 }
+
+func TestWaitPositioningAtx(t *testing.T) {
+	genesis := time.Now()
+	for _, tc := range []struct {
+		desc         string
+		shift, grace time.Duration
+
+		expect string
+	}{
+		{"no wait", 100 * time.Millisecond, 100 * time.Millisecond, ""},
+		{"wait", 100 * time.Millisecond, 0, ""},
+		{"round started", 0, 0, "poet round already started"},
+	} {
+		tc := tc
+		t.Run(tc.desc, func(t *testing.T) {
+			tab := newTestBuilder(t, WithPoetConfig(PoetConfig{
+				PhaseShift:  tc.shift,
+				GracePeriod: tc.grace,
+			}))
+			tab.mclock.EXPECT().CurrentLayer().Return(types.LayerID(0)).AnyTimes()
+			tab.mclock.EXPECT().LayerToTime(gomock.Any()).DoAndReturn(func(lid types.LayerID) time.Time {
+				return genesis.Add(time.Duration(lid) * layerDuration)
+			}).AnyTimes()
```

Comment:
NIT:

```suggestion
			tab.mclock.EXPECT().LayerToTime(gomock.Any()).Return(genesis.Add(time.Duration(lid) * layerDuration)).AnyTimes()
```

Reply:
this should return different results based on lid argument